engine/src/main/java/org/hibernate/search/v6poc/search/dsl/predicate/SpatialWithinPredicateFieldSetContext.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.v6poc.search.dsl.predicate;
+
+import org.hibernate.search.v6poc.backend.spatial.DistanceUnit;
+import org.hibernate.search.v6poc.backend.spatial.GeoPoint;
+import org.hibernate.search.v6poc.backend.spatial.ImmutableGeoPoint;
+import org.hibernate.search.v6poc.util.impl.common.Contracts;
+
+/**
+ * The context used when defining a spatial predicate, after at least one field was mentioned.
+ *
+ * @param <N> The type of the next context.
+ */
+public interface SpatialWithinPredicateFieldSetContext<N> extends MultiFieldPredicateFieldSetContext<SpatialWithinPredicateFieldSetContext<N>> {
+
+	SpatialWithinCirclePredicateContext<N> circle(GeoPoint center, double radiusInMeters);
+
+	default SpatialWithinCirclePredicateContext<N> circle(double latitude, double longitude, double radiusInMeters) {
+		return circle( new ImmutableGeoPoint( latitude, longitude ), radiusInMeters );
+	}
+
+	default SpatialWithinCirclePredicateContext<N> circle(GeoPoint center, double radius, DistanceUnit unit) {
+		Contracts.assertNotNull( radius, "radius" );
+		Contracts.assertNotNull( unit, "unit" );
+
+		return circle( center, unit.toMeters( radius ) );
```

Comment:
I don't know. I like the fact that we are dealing with only one unit that is a standard.

The unit support might vary from one backend to the others, some might only support some of the units or similar.

Reply:
My point was, even if we did forward the unit to the backend, we could just deal with one unit: the backend itself could simply call `unit.toMeter( radius )`. Really not much of a change, but it still allows to do more for some supported units. And even if not all units are supported, backends can always convert to meters to provide "fallback" support.

Reply:
Frankly, I don't really see the point. I prefer having a uniform behavior across all backends. It's not as if converting later could bring exciting features.

There's not 5 ways to do the conversion.

Reply:
The backend itself could perform conversions when it receives meters, though, and use some other unit by default (kilometers for example). Floating point operations being what they are, multiple conversions are not something we should take lightly.

But whatever, if it looks so dumb to you, we'll change it later if the need arises. It's not like it will require API changes.